python/dgl/heterograph.py

```diff
@@ -727,24 +730,41 @@ def ndata(self):
         >>> g = dgl.graph([(0, 1), (1, 2)], 'user', 'follows')
         >>> g.ndata['h'] = torch.zeros(3, 5)
 
+        To set features of all nodes in a heterogeneous graph
+        with multiple node types:
+
+        >>> g = dgl.graph([(0, 1), (1, 2), (1, 0)], 'user', 'like', 'movie')
+        >>> g.ndata['h'] = {'user': torch.zeros(2, 5),
+                            'movie': torch.zeros(3, 5)}
+
         See Also
         --------
         nodes
         """
-        ntid = self.get_ntype_id(None)
-        ntype = self.ntypes[0]
-        return HeteroNodeDataView(self, ntype, ntid, ALL)
+        if len(self.ntypes) == 1:
+            ntid = self.get_ntype_id(None)
+            ntype = self.ntypes[0]
+            return HeteroNodeDataView(self, ntype, ntid, ALL)
+        else:
+            ntids = [self.get_ntype_id(ntype) for ntype in self.ntypes]
+            ntypes = self.ntypes
+            return HeteroNodeDataView(self, ntypes, ntids, ALL)
+
 
     @property
     def srcdata(self):
         """Return the data view of all nodes in the SRC category.
 
         Only works if the graph is either
 
-        * Uni-bipartite and has one node type in the SRC category.
+        * Uni-bipartite
+
+        * Non-uni-bipartite (in this case identical to :any:`DGLHeteroGraph.ndata`)
```

Comment:
I don't think I fully get it. By non-uni-bipartite, do you mean an arbitrary graph other than a bipartite graph?

Reply:
For Uni-bipartite, src and dst are disjoint.
For Non-uni-bipartite, src and dst can have some nodes.

But I think there is no need to highlight these two kinds of graphs.